File: server/routes/questions.js
const express = require('express');
const router = express.Router();
const aiService = require('../services/ai-service');
const ambiguityDetector = require('../services/ambiguity-detector');

// POST /api/questions/generate
router.post('/generate', async (req, res) => {
  try {
    const { text, detectedFlags } = req.body;
    if (!text) {
      return res.status(400).json({ error: 'Text statement is required' });
    }

    let flags = detectedFlags;
    if (!flags) {
      const analysis = ambiguityDetector.analyzeUtterance(text);
      flags = analysis.detectedFlags;
    }

    const question = aiService.generateFallbackQuestion(text, flags);
    res.json({ success: true, data: question });
  } catch (err) {
    res.status(500).json({ success: false, error: err.message });
  }
});

module.exports = router;

File: backend/app/routes/questions.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from ..services.ambiguity_engine import ambiguity_engine

router = APIRouter(tags=["Questions"])

class QuestionGenerateRequest(BaseModel):
    text: Optional[str] = None
    statement: Optional[str] = None
    flags: Optional[List[Dict[str, Any]]] = None

@router.post("/api/questions/generate")
@router.post("/questions/generate")
def generate_realtime_question(payload: QuestionGenerateRequest):
    """
    Generates a targeted clarification question for a live utterance.
    """
    text = payload.text or payload.statement
    if not text or not text.strip():
        raise HTTPException(status_code=400, detail="Missing required 'text' or 'statement' parameter")

    flags, _ = ambiguity_engine.detect_ambiguities_rule_based(text)
    question = ambiguity_engine.generate_fallback_question(text, flags)

    return {
        "success": True,
        "data": question.model_dump()
    }
